base/set.jl

```diff
@@ -191,6 +191,62 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(c)
+
+Remove duplicate items and return the modified collection, as determined by
+[`isequal`](@ref).
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(c::AbstractVector)
+    if (issorted(c) || issorted(c, rev=true)) && !isempty(c)
+        # If c is sorted, then we only need to keep track of one element and add that to c
+        # everytime that we see a new element.
+        count = 1
+        for i in indices(c, 1)
```

Comment:
Should probably use `@inbounds` here? Maybe also in the other branch, not sure.

Also, `indices(c, 1)` could just be `eachindex` since `c` is a vector.